tests/src/analysis/testqgsprocessing.cpp

```diff
@@ -3445,6 +3445,12 @@ void TestQgsProcessing::parameterGeometry()
   QVERIFY( def->checkValueIsAcceptable( QgsReferencedPointXY( QgsPointXY( 1, 2 ), QgsCoordinateReferenceSystem( "EPSG:4326" ) ) ) );
   QVERIFY( def->checkValueIsAcceptable( QgsRectangle( 10, 10, 20, 20 ) ) );
   QVERIFY( def->checkValueIsAcceptable( QgsReferencedRectangle( QgsRectangle( 10, 10, 20, 20 ), QgsCoordinateReferenceSystem( "EPSG:4326" ) ) ) );
+  QVERIFY( def->checkValueIsAcceptable( QString( "MultiPoint((10 10), (20 20))" ) ) );
+
+  // not multipart
+  def->setAllowMultipart( false );
+  QVERIFY( !def->checkValueIsAcceptable( QString( "MultiPoint((10 10), (20 20))" ) ) );
+  def->setAllowMultipart( true );
```

Comment:
Also need to test that the setting persists via calls to clone(), and in the to/fromVariantMap round-trip, and that it's reflected correctly in asPythonString()

Reply:
Just a question: is asScriptCode/fromScriptCode are still relevant methods (since the script fomat has changed in favor of decorators) ?

Reply:
Tests added.

Reply:
Is this possible to backport that PR ?

Reply:
> Just a question: is asScriptCode/fromScriptCode are still relevant methods

Yes - it's used in some of the external processing plugins (eg R provider)